File: frontend/src/app/domain/track.ts
export class Track {

  id: number;
  name: string;
  uploader: any;
  audioUrl: string;
  length: number;

  constructor(id: number, name: string, uploader: any, audioUrl: string, length: number) {
    this.id = id;
    this.name = name;
    this.uploader = uploader;
    this.audioUrl = audioUrl;
    this.length = length;
  }
}

File: core-service/src/main/java/io/aud/coreservice/domain/Track.java
package io.aud.coreservice.domain;

import lombok.*;

import javax.persistence.*;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Entity
public class Track {

    @Id
    @GeneratedValue
    private Long id;

    @Column(nullable = false)
    private String name;

    private String audioUrl;

    private Integer duration;

    @ManyToOne
    private UserAccount uploader;

    @ManyToMany
    private List<Artist> artists;

    @Enumerated(EnumType.STRING)
    private Visibility visibility;

}
